CHANGELOG_UNRELEASED.md

```diff
@@ -105,6 +105,16 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - in `mxpoly.v` new lemmas `horner_mx_diag`, `char_poly_trig`,
    `root_mxminpoly`, and `mxminpoly_diag`
 
+- in `matrix.v`, added `commmx` and `commmxb` the propositional and
```

Comment:
This duplication is really sad. Is this really needed?

Reply:
Unfortunately it is... until we have a structure for abelian categories...

Reply:
Did you mean the duplication between `GRing.comm` and `commmx` ?

Reply:
yep

Reply:
Then yes, `GRing.comm` is bound to (non trivial) rings, and matrices form a (non trivial) ring only when they are square and of manifestly positive size. However during proofs in endomorphism reduction, we take restrictions, which are matrices of size `#|V|` (with `V` a matrix space) and it becomes cumbersome to state commutation between restrictions, unless we relax the setting... and this relaxation corresponds to `commmx`...

Reply:
Is this really needed?

Reply:
Do you see an alternative?

Reply:
You can take a look [here](https://github.com/math-comp/math-comp/pull/207/files#diff-e50ca2313cb816e520933271b7597638) to see the uses cases (do not forget to expand collapsed code blocks)

Reply:
It is difficult to say without knowing the application, naively manipulating matrices of size #|V|.-1.+1 would not work?

Reply:
> It is difficult to say without knowing the application, naively manipulating matrices of size #|V|.-1.+1 would not work?

This is how I started, but since #|V| occurs in many matrix sizes and ordinals in the context and since the subspace is not always assumed to be nonzero, it introduces a nonnegligible overhead in already complicated proofs, and this overhead collapses when introducing  `commmx`.

Reply:
Shouldn't the reason for introducing `commmx` be documented in the header file? This question is likely to be asked again. I think that it is almost a matter of copy-pasting the explanations above and maybe add a link to the file of the development where the issue appears and is solved.

Reply:
Maybe not in the header, but where `commmx` is defined.

Reply:
done